var/spack/repos/builtin/packages/nlopt/package.py

```diff
@@ -60,18 +59,11 @@ def cmake_args(self):
         # Specify on command line to alter defaults:
         # eg: spack install nlopt@master +guile -octave +cxx
 
-        # On is default
-        if "~shared" in spec:
-            args.append("-DBUILD_SHARED_LIBS:Bool=OFF")
-
-        # On is default
-        if "~octave" in spec:
-            args.append("-DNLOPT_OCTAVE:Bool=OFF")
-
-        if "+cxx" in spec:
-            args.append("-DNLOPT_CXX:BOOL=ON")
+        args.append(self.define_from_variant("BUILD_SHARED_LIBS", "shared"))
```

Comment:
then why not immediately `args = [self.define_from_variant(...), ...]`

Reply:
Fixed!